pkg/boot/kexec/memory_linux.go

```diff
@@ -484,13 +479,13 @@ type Memory struct {
 }
 
 // LoadElfSegments loads loadable ELF segments.
-func (m *Memory) LoadElfSegments(r io.ReaderAt) error {
-	f, err := elf.NewFile(r)
+func (m *Memory) LoadElfSegments(r io.ReaderAt) (linux.Object, error) {
+	f, err := linux.ObjectNewFile(r)
```

Comment:
this makes me nervous, and I'm not sure converting other file format to some kind of pseudo-ELF interface is the right abstraction here.

Reply:
can you elaborate on the concerns ? 

This is merely a final design, but sth to begin with. This help commonly get Progs and Entry to kexec target.